.gitlab/datasources/flavors.yaml

```diff
@@ -2,42 +2,107 @@ flavors:
   - name: amd64
     arch: amd64
     alpine: 0
+    fips: 0
     needs_code_checks: true
     needs_layer_publish: true
     suffix: amd64
+    layer_name_base_suffix: ""
+    max_layer_compressed_size_mb: 20
+    max_layer_uncompressed_size_mb: 54
 
   - name: arm64
     arch: arm64
     alpine: 0
+    fips: 0
     needs_code_checks: true
     needs_layer_publish: true
     suffix: arm64
+    layer_name_base_suffix: "-ARM"
+    max_layer_compressed_size_mb: 18
+    max_layer_uncompressed_size_mb: 50
 
   - name: amd64, alpine
     arch: amd64
     alpine: 1
+    fips: 0
     needs_code_checks: false
     needs_layer_publish: false
     suffix: amd64-alpine
 
   - name: arm64, alpine
     arch: arm64
     alpine: 1
+    fips: 0
     needs_code_checks: false
     needs_layer_publish: false
     suffix: arm64-alpine
 
+  - name: amd64, fips
+    arch: amd64
+    alpine: 0
+    fips: 1
+    needs_code_checks: false
+    needs_layer_publish: true
+    suffix: amd64-fips
+    layer_name_base_suffix: "-FIPS"
+    max_layer_compressed_size_mb: 21
+    max_layer_uncompressed_size_mb: 55
+
+  - name: arm64, fips
+    arch: arm64
+    alpine: 0
+    fips: 1
+    needs_code_checks: false
+    needs_layer_publish: true
+    suffix: arm64-fips
+    layer_name_base_suffix: "-FIPS-ARM"
+    max_layer_compressed_size_mb: 19
+    max_layer_uncompressed_size_mb: 51
+
+  - name: amd64, fips, alpine
+    arch: amd64
+    alpine: 1
+    fips: 1
+    needs_code_checks: false
+    needs_layer_publish: false
+    suffix: amd64-fips-alpine
+
+  - name: arm64, fips, alpine
+    arch: arm64
+    alpine: 1
+    fips: 1
+    needs_code_checks: false
+    needs_layer_publish: false
+    suffix: arm64-fips-alpine
+
 # Unfortunately our mutli-arch images don't fit nicely into the flavors
-# structure above.
+# structure above. Since we are making multi-arch images, the suffixes here
+# omit the "arch" component and only include the alpine/fips bits.
 multi_arch_image_flavors:
   - name: basic
     alpine: 0
+    fips: 0
     platform: linux/amd64,linux/arm64
     dependency_names: [amd64, arm64]
     suffix: ""
 
   - name: alpine
     alpine: 1
+    fips: 0
     platform: linux/amd64,linux/arm64
     dependency_names: ["amd64, alpine", "arm64, alpine"]
     suffix: "-alpine"
+
+  - name: fips
+    alpine: 0
+    fips: 1
+    platform: linux/amd64,linux/arm64
+    dependency_names: ["amd64, fips", "arm64, fips"]
+    suffix: "-fips"
```

Comment:
Can't you just do this programatically since you have alpine and fips? Feels unnecessary to add the suffixes here